eng/docgeneration/docindex.yml

```diff
@@ -0,0 +1,119 @@
+resources:
+  repositories:
```

Comment:
Can we please make this yml file a little closer to @scbedd's at https://github.com/Azure/azure-sdk-for-python/blob/master/eng/pipelines/create-doc-index.yml? Such that only things that need to be done to setup the machine are done in the yml and most of the work is done a ps1 script file. That will allow us to run and update this much easier locally. I also like the doc generation stuff under the `doc\` root folder similar to python.

I'm also curious about the naming of the service folder we have https://github.com/Azure/azure-sdk-for-python/blob/master/doc/sphinx/package_service_mapping.json which I think we should have something similar here as well to enable extra configuration.

Reply:
Added this to the Doc Clean Up Issue